a11y/aria-invalid/demos/field-with-aria-invalid/index.html

```diff
@@ -0,0 +1,160 @@
+      font-size: 18px;
+      font-family: "Roboto", sans-serif;
+    }
+
+    .button {
+      display: block;
+      width: 100%;
+      border: 2px solid transparent;
+      border-radius: 6px;
+      padding: 9px 15px;
+      color: #000000;
+      font-size: 18px;
+      font-weight: 300;
+      font-family: inherit;
+      transition: background-color 0.2s linear;
+    }
+
+    .button:hover {
+      background-color: #FFFFFF;
+      cursor: pointer;
+      transition: background-color 0.2s linear;
+    }
+
+    .button:focus-visible {
+      border: 2px solid #FFFFFF;
+      outline: none;
+    }
+
+    .button:focus {
+      border: 2px solid #FFFFFF;
+      outline: none;
+    }
+
+    .button-aqua {
+      background-color: #10F3AF;
+    }
+
+    form {
+      width: 700px;
+      margin-top: 30px;
+    }
+
+    input {
+      width: 100%;
+      margin-bottom: 15px;
+      border: 1px solid #FFFFFF;
+      border-radius: 6px;
+      padding: 10px 15px;
+      background-color: transparent;
+      color: #FFFFFF;
+      font-size: 18px;
+      font-weight: 300;
+      font-family: inherit;
+      -webkit-appearance: none;
+      appearance: none;
+    }
+
+    input:focus {
+      border-color: #10F3AF;
+      outline: none;
+    }
+
+    input[aria-invalid=true] {
+      border-color: #F498AD;
+    }
+
+    label {
+      margin-right: 55px;
+      font-size: 24px;
+      font-weight: 500;
+      line-height: 1;
+    }
+
+    .form-row {
+      display: flex;
+      justify-content: space-between;
+    }
+
+    .container {
+      width: 350px;
+    }
+
+    .input-container {
+      margin-bottom: 25px;
+    }
+
+    @media (max-width: 768px) {
+      body {
+        padding: 30px;
+      }
+
+      label {
+        display: inline-block;
+        margin: 0 0 10px;
+      }
+
+      .form-row {
+        display: block;
+      }
+
+      form, .container {
+        width: 100%;
+      }
+  </style>
+</head>
+<body>
+  <div class="form-row">
+    <label for="drink">Любимый напиток:</label>
+    <div class="container">
+      <div class="input-container">
+        <input id="drink" type="text" name="drink" aria-invalid="false" aria-describedby="error">
+        <span class="error-message" id="error" aria-live="assertive"></span>
+      </div>
+      <button id="button" class="button button-aqua">Рассказать</button>
+    </div>
+  </div>
+  <script>
+    const input = document.getElementById("drink");
+    const button = document.getElementById("button");
+    const error = document.getElementById("error");
+    const errorText = document.createTextNode("Введите один любимый напиток. Например, чай, кофе или вода.");
+
+    button.addEventListener("click", e => {
+      const value = input.value;
+
+      if (!value) {
+        input.setAttribute("aria-invalid", "true");
+        error.appendChild(errorText);
+      } else {
+        input.setAttribute("aria-invalid", "false");
+        error.innerHTML = "";
+      }
+    });
+  </script>
```

Comment:
```suggestion
  <script>
    const input = document.getElementById('drink')
    const button = document.getElementById('button')
    const error = document.getElementById('error')
    const errorText = document.createTextNode('Введите один любимый напиток. Например, чай, кофе или вода.')

    button.addEventListener('click', event => {
      const value = input.value

      if (!value) {
        input.setAttribute('aria-invalid', 'true')
        error.appendChild(errorText)
      } else {
        input.setAttribute('aria-invalid', 'false')
        error.innerHTML = ''
      }
    })
  </script>
```

Отредачила JS-код по нашим правилам. Прости, если раньше не обращала на это внимание.

Reply:
А, буду иметь теперь в виду!